bokehjs/src/lib/core/visuals.ts

```diff
@@ -8,6 +8,166 @@ import {LineJoin, LineCap, FontStyle, TextAlign, TextBaseline} from "./enums"
 import {HasProps} from "./has_props"
 import {ColumnarDataSource} from "models/sources/columnar_data_source"
 
+function _horz(ctx: CanvasRenderingContext2D, h: number, h2: number): void {
```

Comment:
`Context2d` from `core/util/canvas` would be a better choice, especially if you would like to use any features we extend the native context with.